openc3/lib/openc3/packets/packet.rb

```diff
@@ -872,6 +876,33 @@ def define_reserved_items
       item.description = 'OpenC3 packet received count'
     end
 
+    # Reset the packet to just derived items
+    def clear_all_non_derived_items
+      @defined_length = 0
+      @defined_length_bits = 0
+      @pos_bit_size = 0
+      @neg_bit_size = 0
+      @fixed_size = true
+      @short_buffer_allowed = false
+      @id_items = nil
+      @limits_items = nil
+      new_items = {}
+      new_sorted_items = []
+      @items.each do |name, item|
+        if item.data_type == :DERIVED
+          new_items[name] = item
+        end
+      end
+      @sorted_items.each do |item|
+        if item.data_type == :DERIVED
+          new_sorted_items << item
+        end
+      end
+      @items = new_items
+      @sorted_items = new_sorted_items
+      clear_config_name()
+    end
```

Comment:
I don't see this being called

Reply:
Its called by the user.  See my prometheus plugin soon.